### PR TITLE
Interactivity API: Backport fixes from Core

### PR DESCRIPTION
**Pull request by cbravobernal**
## What?

Backports fixes done in Core to the Gutenberg compat folder:

- https://github.com/WordPress/wordpress-develop/pull/6267/
- https://github.com/WordPress/wordpress-develop/pull/6257/
- https://github.com/WordPress/wordpress-develop/pull/6245/
- https://github.com/WordPress/wordpress-develop/pull/6251/
- https://github.com/WordPress/wordpress-develop/pull/6261

### CODE REVIEW

**Review by SantosGuillamot — APPROVED** [review]
LGTM! Thanks for taking care of that 🙂

File: lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api.php [hunk 1]
```diff
@@ -572,15 +601,21 @@ private function data_wp_bind_processor( WP_Interactivity_API_Directives_Process
 					$attribute_value = $p->get_attribute( $attribute_name );
 					$result          = $this->evaluate( $attribute_value, end( $namespace_stack ), end( $context_stack ) );
 
-					if ( null !== $result && ( false !== $result || '-' === $bound_attribute[4] ) ) {
+					if ( null !== $result && (
+					false !== $result ||
+					( strlen( $bound_attribute ) > 5 && '-' === $bound_attribute[4] )
```
Comment (sirreal) [inline lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api.php]:
```suggestion
						false !== $result ||
						( strlen( $bound_attribute ) > 5 && '-' === $bound_attribute[4] )
```

File: lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api.php [hunk 2]
```diff
@@ -572,15 +601,21 @@ private function data_wp_bind_processor( WP_Interactivity_API_Directives_Process
...
 						 * If the result of the evaluation is a boolean and the attribute is
 						 * `aria-` or `data-, convert it to a string "true" or "false". It
 						 * follows the exact same logic as Preact because it needs to
 						 * replicate what Preact will later do in the client:
 						 * https://github.com/preactjs/preact/blob/ea49f7a0f9d1ff2c98c0bdd66aa0cbc583055246/src/diff/props.js#L131C24-L136
 						 */
-						if ( is_bool( $result ) && '-' === $bound_attribute[4] ) {
+						if (
+						is_bool( $result ) &&
+						( strlen( $bound_attribute ) > 5 && '-' === $bound_attribute[4] )
```
Comment (sirreal) [inline lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api.php]:
```suggestion
							is_bool( $result ) &&
							( strlen( $bound_attribute ) > 5 && '-' === $bound_attribute[4] )
```

**Review by sirreal — APPROVED** [review]
I reviewed these ports and the changes look correct. Thanks!

- https://github.com/WordPress/wordpress-develop/pull/6257
- https://github.com/WordPress/wordpress-develop/pull/6261

**Comment by cbravobernal** [comment]
@sirreal I'm removing the backport label as is used to indicate that the backport is needed, not done. 😄

**Pull request merged**